File: test-samples/crypto_example.js
// Sample JavaScript file with cryptographic usage for testing PQ-check

const crypto = require('crypto');
const jwt = require('jsonwebtoken');

// RSA key generation (quantum-vulnerable)
crypto.generateKeyPairSync('rsa', {
  modulusLength: 2048,
  publicKeyEncoding: { type: 'spki', format: 'pem' },
  privateKeyEncoding: { type: 'pkcs8', format: 'pem' }
});

// ECDH key exchange (quantum-vulnerable)
const ecdh = crypto.createECDH('secp256k1');
ecdh.generateKeys();

// Digital signature (check algorithm)
const sign = crypto.createSign('SHA256');

// JWT with RSA (quantum-vulnerable)
const token = jwt.sign({ user: 'test' }, 'secret', { algorithm: 'RS256' });

// AES encryption (quantum-resistant - good!)
const cipher = crypto.createCipheriv('aes-256-gcm', key, iv);

File: test-samples/crypto_example.py
# Sample Python file with cryptographic usage for testing PQ-check

from cryptography.hazmat.primitives.asymmetric import rsa
from cryptography.hazmat.primitives.asymmetric import ec
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.backends import default_backend
import jwt

# RSA key generation (quantum-vulnerable)
private_key = rsa.generate_private_key(
    public_exponent=65537,
    key_size=2048,
    backend=default_backend()
)

# EC key generation (quantum-vulnerable)
ec_private_key = ec.generate_private_key(ec.SECP256R1(), default_backend())

# JWT with RSA (quantum-vulnerable)
token = jwt.encode({"user": "test"}, "secret", algorithm="RS256")

# AES encryption (quantum-resistant - good!)
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
